internal/commands/image/list.go

```diff
@@ -70,47 +78,82 @@ func getPackages(bundleStore store.BundleStore, references []reference.Named) ([
 	return packages, nil
 }
 
-func printImages(dockerCli command.Cli, refs []pkg) error {
+func printImages(dockerCli command.Cli, options imageListOptions, refs []pkg) error {
 	w := tabwriter.NewWriter(dockerCli.Out(), 0, 0, 1, ' ', 0)
 
-	printHeaders(w)
+	columns := getColumns(options)
+	printHeaders(w, columns)
 	for _, ref := range refs {
-		printValues(w, ref)
+		printValues(w, columns, ref)
 	}
 
 	return w.Flush()
 }
 
-func printHeaders(w io.Writer) {
+func getColumns(options imageListOptions) columns {
+	if options.showDigests {
+		return withDigestColumns
+	}
+	return defaultColumns
+}
+
+func printHeaders(w io.Writer, cols columns) {
 	var headers []string
-	for _, column := range listColumns {
+	for _, column := range cols {
 		headers = append(headers, column.header)
 	}
 	fmt.Fprintln(w, strings.Join(headers, "\t"))
 }
 
-func printValues(w io.Writer, ref pkg) {
+func printValues(w io.Writer, cols columns, ref pkg) {
 	var values []string
-	for _, column := range listColumns {
+	for _, column := range cols {
 		values = append(values, column.value(ref))
 	}
 	fmt.Fprintln(w, strings.Join(values, "\t"))
 }
 
 var (
-	listColumns = []struct {
-		header string
-		value  func(p pkg) string
-	}{
+	defaultColumns = []column{
 		{"APP IMAGE", func(p pkg) string {
 			return reference.FamiliarString(p.ref)
 		}},
 		{"APP NAME", func(p pkg) string {
 			return p.bundle.Name
 		}},
 	}
+
+	withDigestColumns = columns{
+		{"REPOSITORY", func(p pkg) string {
```

Comment:
Why do we have `REPOSITORY` and `TAG` here, and not just `APP IMAGE` and `APP NAME` ? 🤔